tests/cli/test_run.py

```diff
@@ -262,9 +262,124 @@ def test_main_cannot_write_zuliprc_given_good_credentials(
+        return zuliprc_path
+    return func
+
+
+@pytest.mark.parametrize("config_key,config_value,footlinks_output", [
+        ("footlinks", "disabled",
+            "   maximum footlinks value '0' "
+            "specified in zuliprc file from footlinks."),
+        ("footlinks", "enabled",
+            "   maximum footlinks value '3' "
+            "specified in zuliprc file from footlinks."),
+        ("maximum-footlinks", "3",
+            "   maximum footlinks value '3' specified in zuliprc file."),
+        ("maximum-footlinks", "0",
+            "   maximum footlinks value '0' specified in zuliprc file.")
+    ],
+    ids=[
+        "footlinks_disabled",
+        "footlinks_enabled",
+        "maximum-footlinks_3",
+        "maximum-footlinks_0"
+    ])
+def test_successful_main_function_with_config(
+        config_key, config_value,
+        capsys, mocker, parameterized_zuliprc,
+        footlinks_output
+                                             ):
+    config = {
+        "theme": "default",
+        "autohide": "autohide",
+        "notify": "enabled",
+        "color-depth": "256"
+    }
+    config[config_key] = config_value
+    zuliprc = parameterized_zuliprc(config)
+    mocker.patch("zulipterminal.core.Controller.__init__",
+                 return_value=None)
+    mocker.patch("zulipterminal.core.Controller.main",
+                 return_value=None)
+    with pytest.raises(SystemExit):
+        main(["-c", zuliprc])
+
+    captured = capsys.readouterr()
+    lines = captured.out.strip().split("\n")
+    expected_lines = [
+        'Loading with:',
+        "   theme 'zt_dark' specified in zuliprc file (by alias 'default').",
+        "   autohide setting 'autohide' specified in zuliprc file.",
+        footlinks_output,
+        "   color depth setting '256' specified in zuliprc file."
+                     ]
+    assert lines == expected_lines
+
+
+@pytest.mark.parametrize("zulip_config", [
+    {
+        "footlinks": "enabled",
+        "maximum-footlinks": "3"
+    }])
+def test_successful_main_function_with_footlinks_and_maximum_footlinks(
+        parameterized_zuliprc, zulip_config, capsys,
+        mocker
+                                                                      ):
+    zuliprc = parameterized_zuliprc(zulip_config)
+    mocker.patch("zulipterminal.core.Controller.__init__",
+                 return_value=None
+                 )
+    mocker.patch("zulipterminal.core.Controller.main",
+                 return_value=None)
+    with pytest.raises(SystemExit) as e:
+        main(["-c", zuliprc])
+        assert str(e.value) == "1"
+    captured = capsys.readouterr()
+    lines = captured.out.strip()
+    expected_lines = "\033[91m{}\033[0m".format(
+        "Footlinks property is not allowed alongside maximum-footlinks")
+    assert lines == expected_lines
+
+
+@pytest.mark.parametrize("zulip_config", [
+        {"maximum-footlinks": "-3"}
+    ])
+def test_minimum_footlinks(
```

Comment:
The body of these two tests looks similar enough to combine through parametrization?

Maybe a test name starting `test_main_error_`? This isn't really a success we're testing for :)